pkg/registry/apis/datasource/register.go

```diff
@@ -224,7 +227,9 @@ func (b *DataSourceAPIBuilder) getPluginContext(ctx context.Context, uid string)
 	if err != nil {
 		return backend.PluginContext{}, err
 	}
-	return b.contextProvider.PluginContextForDataSource(ctx, instance)
+	pctx, err := b.contextProvider.PluginContextForDataSource(ctx, instance)
+	pctx.APIVersion = b.connectionResourceInfo.APIVersion()
```

Comment:
This can probably be in a different PR? It's unrelated to having apiserver info in the plugin DTO.

Reply:
👍🏻 -- i can split it out soon, already a few too many open PRs for hte same topic 🤣